src/test/java/com/google/api/generator/engine/ast/ReferenceConstructorExprTest.java

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+import static org.junit.Assert.assertThrows;
+
+import org.junit.Test;
+
+public class ReferenceConstructorExprTest {
+
+  @Test
+  public void validReferenceConstructorExpr_thisConstructorBasic() {
+    VaporReference ref =
+        VaporReference.builder()
+            .setName("Student")
+            .setPakkage("com.google.example.examples.v1")
+            .build();
+    TypeNode typeNode = TypeNode.withReference(ref);
+    ReferenceConstructorExpr.thisBuilder().setType(typeNode).build();
+    // No exception thrown, so we succeeded.
+  }
+
+  @Test
+  public void validReferenceConstructorExpr_superConstructorBasic() {
+    VaporReference ref =
+        VaporReference.builder()
+            .setName("Student")
+            .setPakkage("com.google.example.examples.v1")
+            .build();
+    TypeNode typeNode = TypeNode.withReference(ref);
+    ReferenceConstructorExpr.superBuilder().setType(typeNode).build();
+    // No exception thrown, so we succeeded.
+  }
+
+  @Test
+  public void invalidReferenceConstructorExpr_nonReferenceType() {
+    assertThrows(
+        IllegalStateException.class,
+        () -> {
+          ReferenceConstructorExpr.thisBuilder().setType(TypeNode.BOOLEAN).build();
+        });
+  }
```

Comment:
Add a test for `Boolean` or `Integer` object, e.g. for classes that may inherit from those objects.

Reply:
Sorry, I misunderstand here. First `Boolean` and `Integer` are final class which can't be inherited.

Do you mean add test which setType(TypeNode.Integer)? For now, it will not throw error.
In `ReferenceConstructorExpr.java`, only checked the type `isReferencedType`, however, `Integer, String` are referenceType but they are belongs to `java.lang`

If we want the check, we could add 
```
      Preconditions.checkState(referenceConstructorExpr.type().reference().isFromPackage(PKG_JAVA_LANG),
          "ReferenceConstructExpr can only refer to custom object. ");
``

Let me know what you think. @miraleung thanks.

Reply:
SG, let's leave it as is then.